components/ca-treeview.js

```diff
@@ -0,0 +1,263 @@
+        }
+
+        /**
+         * Sets the data
+         * @param {array} value, array to set as the data, defaults to empty. Invokes render.
+         */
+        set data(value = []) {
+            this._data = value;
+            this.render();
+        }
+
+        /**
+         * Gets selected item
+         * @returns {HTMLElement} selected Item
+         */
+        get selected() {
+            return this._selected;
+        }
+
+        /**
+         * Sets the selected item.
+         * @param {HTMLElement} el current element.
+         * @returns {int} max-children.
+         */
+        set selected(el) {
+            this._selected = el;
+        }
+
+        /**
+         * Gets the max children allowed.
+         * @returns {int} max-children.
+         */
+        get maxChildren() {
+            return parseInt(this.getAttribute('data-max-children') || '0', 10);
+        }
+
+        /**
+         * Sets the max children allowed.
+         * @param {int} value number of allow childre, defaults to 0.
+         */
+        set maxChildren(value = 0) {
+            this.setAttribute('data-max-children', parseInt(value, 10));
+        }
+
+        /**
+         * Load the data via an XHR request.
+         * @param {HTMLElement} parent to render with data.
+         * @param {string} dataSrc, url to load data form.
+         * @returns {undefined} either data is loaded or render is called.
+         */
+        loadData(parent, dataSrc) {
+            const dataSource = dataSrc || this.dataSrc;
+            ajax
+                .execute({ url: dataSource, dataType: 'json' })
+                .then(data => {
+                    // if this is a schema with instance links, grab the instance url
+                    const instancesUrl = (data.$schema && data.items.links) ? (data.items.links.find(link => link.rel === 'instances') || {}).href : '';
+                    if (instancesUrl !== '') {
+                        // we need to go get the instance data and recall this function to trigger rendering
+                        this.loadData(parent, instancesUrl);
+                    } else {
+                        createElement(this.ol);
+                        this.render(parent, data);
+                    }
+                });
+        }
+
+        /**
+         * create a child within the tree view
+         * @param {HTMLElement} parent, append to this.
+         * @param {string} tagName, the html tag to create.
+         * @param {object} item, schema object.
+         * @returns {HTMLElement} the newly created list item.
+         */
+        createChild(parent, tagName, item) {
+            const label = item.label || item.title || 'undefined';
+            const rel = (item.rel || '').toLowerCase().replace(/(describedby|described_by):/gi, '');
+
+            // get a list of attributes to be added to the element
+            const attributes = Object.keys(item).reduce((attributeList, key) => {
```

Comment:
This would ban it for all projects that lint using airbnb as a base, it can be overridden here https://github.com/cambridgeweblab/eslint-config-weblab/blob/master/overrides.js, though most projects aren't using this common package, yet.

I agree... I think their bias is coming in here. But in their defence, they did label it as `good` rather than `best`, and their best examples are pretty great. So what do we do here - revert and update the eslint rules or chain array functions?